### PR TITLE
Adjust export of enum traversal operations

### PR DESCRIPTION
**Pull request by simonz-bq**
Adjusted the exporting of traversal operations that were type aliases of string enums to be closer to the GLV in ohter languages.

### CODE REVIEW

**Comment by lyndonbauto** [comment]
Thanks for the change @simonz-bq. The actual change itself looks great, this will make our syntax more natural for users who are used to using things like `Scope.Local` (as opposed to our previous just `gremlingo.Local`).

I noticed:

```
gremlin-go-integration-tests    | driver/cucumber/gremlin.go:35: undefined: gremlingo.Any
gremlin-go-integration-tests    | driver/cucumber/gremlin.go:36: undefined: gremlingo.None
gremlin-go-integration-tests    | driver/cucumber/gremlin.go:41: undefined: gremlingo.None
gremlin-go-integration-tests    | driver/cucumber/gremlin.go:44: undefined: gremlingo.None
gremlin-go-integration-tests    | driver/cucumber/gremlin.go:47: undefined: gremlingo.Value
gremlin-go-integration-tests    | driver/cucumber/gremlin.go:47: undefined: gremlingo.Asc
gremlin-go-integration-tests    | driver/cucumber/gremlin.go:48: undefined: gremlingo.Label
gremlin-go-integration-tests    | driver/cucumber/gremlin.go:48: undefined: gremlingo.Id
gremlin-go-integration-tests    | driver/cucumber/gremlin.go:62: undefined: gremlingo.Id
gremlin-go-integration-tests    | driver/cucumber/gremlin.go:65: undefined: gremlingo.Label
```

So I think you will need to update the translator to have `gremlingo.T.Label` as opposed to `gremlingo.Label`.

Let me know if you need help with this as that section of code is a kind of esoteric.

File: gremlin-core/src/test/java/org/apache/tinkerpop/gremlin/process/traversal/translator/GolangTranslatorTest.java [hunk 1]
```diff
@@ -88,7 +88,7 @@ public void shouldTranslateGoNamedSteps() {
         assertEquals("g.V().Has(\"person\", \"name\", \"marko\").Where(gremlingo.T__.OutE().Count().Is(2).And(gremlingo.T__.Not(gremlingo.T__.InE().Count().Is(3))))", gremlinAsGo);
     }
 
-    // TODO AN-987: TraversalStrategy implementation in Gremlin-go
+    // TODO: TraversalStrategy implementation in Gremlin-go
```
Comment (lyndonbauto) [inline gremlin-core/src/test/java/org/apache/tinkerpop/gremlin/process/traversal/translator/GolangTranslatorTest.java:91]: This is actually done.
```suggestion
```

File: gremlin-go/driver/traversal.go [hunk 2]
```diff
@@ -124,128 +124,187 @@ func (t *Traversal) GetResultSet() (ResultSet, error) {
...
 
+type barrier string
+
+type barriers struct {
+	NormSack barrier
+}
+
 // Barrier is any step that requires all left traversers to be processed prior to emitting result traversers to the right.
-type Barrier string
+var Barrier = barriers{
```
Comment (lyndonbauto) [inline gremlin-go/driver/traversal.go:134]: Missing a space after barriers, might need to `go fmt`

File: gremlin-core/src/test/java/org/apache/tinkerpop/gremlin/process/traversal/translator/GolangTranslatorTest.java (hunk 1 above)
Comment (simonz-bq) [reply gremlin-core/src/test/java/org/apache/tinkerpop/gremlin/process/traversal/translator/GolangTranslatorTest.java]: Added a tracker for the missing translator logic. The implementation of strategies is done but aren't able to be translated yet.

**Review by lyndonbauto — APPROVED** [review]

**Pull request merged**